src/inet/InetInterface.cpp

```diff
@@ -897,6 +914,45 @@ CHIP_ERROR InterfaceId::GetLinkLocalAddr(IPAddress * llAddr) const
     return (found) ? CHIP_NO_ERROR : INET_ERROR_ADDRESS_NOT_FOUND;
 }
 
+BitFlags<InterfaceAddressIterator::Flags> InterfaceAddressIterator::GetFlags()
+{
+    BitFlags<InterfaceAddressIterator::Flags> flags;
+
+#if __linux__
+    if (!HasCurrent())
+    {
+        return flags;
+    }
+
+    if (mCurAddr->ifa_flags & IFA_F_OPTIMISTIC)
+    {
+        flags.Set(InterfaceAddressIterator::Flags::kNotFinal);
+    }
+
+    if (mCurAddr->ifa_flags & IFA_F_DADFAILED)
+    {
+        flags.Set(InterfaceAddressIterator::Flags::kNotFinal);
+    }
+
+    if (mCurAddr->ifa_flags & IFA_F_TENTATIVE)
+    {
+        flags.Set(InterfaceAddressIterator::Flags::kNotFinal);
+    }
```

Comment:
Suggest to ifdef those flasg separately in case some platforms like Android don't have all of them

Reply:
I would expect CI to detect this. I believe for android we care about IFA_F_OPTIMISTIC and DADFAILED, however the flags kNotFinal and such can remain the same as if a platform does not have that concept, the flags are not set. This is what currently happen for all non-linux